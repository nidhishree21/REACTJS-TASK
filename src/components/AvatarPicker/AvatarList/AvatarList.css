ul {
  padding: 0;
}

ul li {
  display: inline-block;
  margin: 7px;
  list-style: none;
}
