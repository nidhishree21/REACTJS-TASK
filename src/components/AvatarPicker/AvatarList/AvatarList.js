import React from "react";
import PropTypes from "prop-types";
import Avatar from "../Avatar/Avatar";
import "./AvatarList.css";

const AvatarList = props => {
  return (
    <ul>
      {props.avatars.map(avatar => {
        return (
          <li key={avatar.id}>
            <Avatar
              avatar={avatar}
              onClickAvatar={props.onClickAvatar}
              isActive={props.isActive}
              highlight={props.highlight(avatar.id)}
              showSpinner={props.showSpinner(avatar.id)}
            />
          </li>
        );
      })}
    </ul>
  );
};

AvatarList.propTypes = {
  avatars: PropTypes.arrayOf(PropTypes.object).isRequired,
  onClickAvatar: PropTypes.func.isRequired,
  isActive: PropTypes.bool.isRequired,
  highlight: PropTypes.func.isRequired,
  showSpinner: PropTypes.func.isRequired
};

export default AvatarList;
