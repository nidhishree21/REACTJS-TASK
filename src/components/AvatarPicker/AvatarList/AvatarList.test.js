import React from "react";

import { mount } from "enzyme";
import AvatarList from "./AvatarList";

const testProps = {
  avatars: [
    {
      src: "/static/media/avatar1.7467b519.png",
      label: "Avatar 1",
      id: 1
    },
    {
      src: "/static/media/avatar3.c1a8b069.png",
      label: "Avatar 2",
      id: 2
    }
  ],
  onClickAvatar: jest.fn(() => {}),
  isActive: false,
  highlight: jest.fn(() => {
    return false;
  }),
  showSpinner: jest.fn(() => {
    return false;
  })
};

describe("AvatarList", () => {
  const wrapper = mount(<AvatarList {...testProps} />);
  it("renders the Avatar List correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("calls onClickAvatar function on clicking an avatar in the list", () => {
    const listAvatar = wrapper.find("[id=2]");
    listAvatar.simulate("click");
    expect(testProps.onClickAvatar).toHaveBeenCalledTimes(1);
  });

  it("returns correct number of list items", () => {
    expect(wrapper.find("ul").children().length).toBe(2);
  });

  it("calls highlight", () => {
    expect(testProps.highlight).toHaveBeenCalledTimes(2);
  });

  it("calls showspinner", () => {
    expect(testProps.showSpinner).toHaveBeenCalledTimes(2);
  });
});
