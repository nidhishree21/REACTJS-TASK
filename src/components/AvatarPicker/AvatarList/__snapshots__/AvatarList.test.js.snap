// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarList renders the Avatar List correctly 1`] = `
<AvatarList
  avatars={
    Array [
      Object {
        "id": 1,
        "label": "Avatar 1",
        "src": "/static/media/avatar1.7467b519.png",
      },
      Object {
        "id": 2,
        "label": "Avatar 2",
        "src": "/static/media/avatar3.c1a8b069.png",
      },
    ]
  }
  highlight={[Function]}
  isActive={false}
  onClickAvatar={[Function]}
  showSpinner={[Function]}
>
  <ul>
    <li
      key="1"
    >
      <Avatar
        avatar={
          Object {
            "id": 1,
            "label": "Avatar 1",
            "src": "/static/media/avatar1.7467b519.png",
          }
        }
        highlight={false}
        isActive={false}
        onClickAvatar={[Function]}
        showSpinner={false}
      >
        <div
          className="ListImgWrapper null"
          id={1}
          onClick={[Function]}
          onKeyUp={[Function]}
          tabIndex="0"
        >
          <img
            alt="Avatar 1"
            className="Img"
            src="/static/media/avatar1.7467b519.png"
          />
        </div>
      </Avatar>
    </li>
    <li
      key="2"
    >
      <Avatar
        avatar={
          Object {
            "id": 2,
            "label": "Avatar 2",
            "src": "/static/media/avatar3.c1a8b069.png",
          }
        }
        highlight={false}
        isActive={false}
        onClickAvatar={[Function]}
        showSpinner={false}
      >
        <div
          className="ListImgWrapper null"
          id={2}
          onClick={[Function]}
          onKeyUp={[Function]}
          tabIndex="0"
        >
          <img
            alt="Avatar 2"
            className="Img"
            src="/static/media/avatar3.c1a8b069.png"
          />
        </div>
      </Avatar>
    </li>
  </ul>
</AvatarList>
`;
